cargo/cargo_build_script_runner/lib.rs

```diff
@@ -103,12 +103,17 @@ impl BuildScriptOutput {
     /// Converts a [BufReader] into a vector of [BuildScriptOutput] enums.
     fn outputs_from_reader<T: Read>(mut reader: BufReader<T>) -> Vec<BuildScriptOutput> {
         let mut result = Vec::<BuildScriptOutput>::new();
-        let mut line = String::new();
-        while reader.read_line(&mut line).expect("Cannot read line") != 0 {
-            if let Some(bso) = BuildScriptOutput::new(&line) {
+        let mut buf = Vec::new();
+        while reader
+            .read_until(b'\n', &mut buf)
```

Comment:
Could you add a test that if the output doesn't end with a trailing newline, we still parse the last line of output? (i.e. can you remove the trailing newline from the example output in the test?)

Also, could you add a line of invalid utf-8 to the testdata in the test, to verify this change?

Reply:
Done